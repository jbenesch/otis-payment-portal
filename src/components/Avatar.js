import styled from 'styled-components'
import PropTypes from 'prop-types'

const Avatar = styled.img`
  border-radius: 50%;
  width: ${props => {
    if (props.size) {
      switch (props.size) {
        case 'large':
          return '120px'
        case 'small':
          return '30px'
        case 'medium':
        default:
          return '50px'
      }
    }
    return '50px'
  }};
`

Avatar.propTypes = {
  size: PropTypes.oneOf(['small', 'medium', 'large'])
}

Avatar.defaultProps = {
  size: 'large'
}

export default Avatar
