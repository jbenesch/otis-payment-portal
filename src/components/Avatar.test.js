import React from 'react';
import renderer from 'react-test-renderer';
import Avatar from './Avatar';

it('matches a previous snapshot', () => {
  const component = renderer.create(
    <Avatar src="https://randomuser.me/api/portraits/men/51.jpg" size="large" />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
