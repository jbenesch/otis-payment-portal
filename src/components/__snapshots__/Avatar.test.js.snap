// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches a previous snapshot 1`] = `
<img
  className="sc-bdVaJa hrdBvl"
  size="large"
  src="https://randomuser.me/api/portraits/men/51.jpg"
/>
`;
